src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/CvlCaseDto.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceKind
+import java.time.LocalDate
+
+data class CvlCaseDto(
+  val nomisId: String,
+  val licenceStartDate: LocalDate? = null,
+  val isEligible: Boolean = false,
+  val eligibleKind: LicenceKind? = null,
+  val ineligiblityReasons: EligibilityAssessment,
+)
+
+data class EligibilityAssessment(
+  val genericIneligibilityReasons: List<String> = emptyList(),
+  val crdIneligibilityReasons: List<String> = emptyList(),
+  val prrdIneligibilityReasons: List<String> = emptyList(),
+  val isEligible: Boolean,
+  val eligibleKind: LicenceKind? = when (isEligible) {
+    true if crdIneligibilityReasons.isEmpty() -> LicenceKind.CRD
+    true if prrdIneligibilityReasons.isEmpty() -> LicenceKind.PRRD
```

Comment:
Get a warning on these: 
```The feature "when guards" is experimental and should be enabled explicitly. This can be done by supplying the compiler argument '-Xwhen-guards', but note that no stability guarantees are provided```